### PR TITLE
Add instructions on how to use pre-commit tool

### PR DESCRIPTION
**Pull request by MarkoSagadin**
## Description

As the title says. 

I imagine that the added text serves as a material for the engineer to get acquainted with the `pre-commit` tool. The link to it will be added to the `irnas-zephyr-template` repository.

## Areas of interest for the reviewer

Are instructions clear?
Should I explain some other specific hook?

## After-review steps

- I will merge PR by myself.

[style guidelines]:
  https://github.com/IRNAS/irnas-guidelines-docs/blob/main/docs/developer_guidelines.md

### CODE REVIEW

File: tools/pre-commit/README.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+## Usage
+
+### Repository setup
+
+To make `pre-commit` run before every `git commit` you need to install it into
+the repository:
+
+```bash
+pre-commit install
```
Comment (herrjade) [inline tools/pre-commit/README.md:38]: doesn't this require install via pip first ?

File: tools/pre-commit/README.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+[Git hook]: https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks
+
+A `.pre-commit-commit.yaml` file, located in the project root dir, defines
+exactly which hooks should run, with which arguments, and when.
+
+See our `.pre-commit-commit.yaml` file in the [irnas-zephyr-template]
+repository.
+
+[irnas-zephyr-template]: https://github.com/IRNAS/irnas-zephyr-template
+
```
Comment (TjazVracko) [inline tools/pre-commit/README.md:21]: I suggest you add something like the following here:

```suggestion
In essence, pre-commit will not allow you to create a commit until the codebase follows our guidelines.
```

File: tools/pre-commit/README.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+```
+
+## Expected workflow and fixing errors
+
+Developer's workflow with the `pre-commit` looks something like this:
+
+1. You made some changes to the repository and you have added them with
+   `git add` to the staging area.
+2. You run `git commit` to commit the files.
+3. The `pre-commit` tool runs all hooks on your staged files.
```
Comment (TjazVracko) [inline tools/pre-commit/README.md:81]:
```suggestion
3. The `pre-commit` tool automatically runs all hooks on your staged files and checks your commit message.
```

File: tools/pre-commit/README.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+   ```
+
+   In other cases where some error is detected, the `pre-commit` will print it
+   out.
+
+5. You fix the error, add newly modified files, run `git commit` and see
+   `pre-commit` hooks run again.
+6. Repeat this until all errors are resolved, at that point, the commit is
+   accepted and created.
+
```
Comment (TjazVracko) [inline tools/pre-commit/README.md:101]:
```suggestion

## Hooks

The following section describes some of the hooks we are using which might require more knowledge from the developers. 

```

File: tools/pre-commit/README.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+pre-commit run --all-files clang-format
+```
+
+### Testing commit messages
+
+To test if your commit message passes the rules imposed by the `committed` tool:
+
+```bash
+pre-commit run --hook-stage commit-msg committed --commit-msg-filename <file_with_commit_message>
+```
```
Comment (TjazVracko) [inline tools/pre-commit/README.md:72]: I would move this to the end of the file into a _Testing and debugging_ section

**Review by TjazVracko — CHANGES_REQUESTED** [review]
Good start.

Since most of the hooks can be configured using special files or directly in `.pre-commit-commit.yaml`

I think you need to mention this. Especially for formatters - where I might want to exclude folders or files from being formatted.

**Review by herrjade — APPROVED** [review]

File: tools/pre-commit/README.md (hunk 5 above)
Comment (MarkoSagadin) [reply tools/pre-commit/README.md:84]: Moved.

**Comment by MarkoSagadin** [comment]
* **#38** <a href="https://app.graphite.dev/github/pr/IRNAS/irnas-guidelines-docs/38?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#37** <a href="https://app.graphite.dev/github/pr/IRNAS/irnas-guidelines-docs/37?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @MarkoSagadin and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: tools/pre-commit/README.md [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+
+For example:
+
+```bash
+pre-commit run --all-files clang-format
+```
+
+### Testing commit messages
+
+To test if your commit message passes the rules imposed by the `committed` tool:
```
Comment (TjazVracko) [inline tools/pre-commit/README.md:198]:
```suggestion
To test if your commit message passes the rules imposed by the `committed` tool, first add your commit message to a file `<msg.txt>`, then run:
```

**Review by TjazVracko — CHANGES_REQUESTED** [review]

File: tools/pre-commit/README.md (hunk 6 above)
Comment (MarkoSagadin) [reply tools/pre-commit/README.md]: Added

**Pull request merged**